### ISSUE TITLE
Wrong DateTime modify handling

### ISSUE THREAD

**Issue by MartinMajor** [facebook/hhvm#4598]
```php
$datetime = new DateTime('2015-01-08');
$datetime->modify('first day of this month');
echo $datetime->format('Y-m-d');
```

expected result and result with PHP 5.5:
> 2015-01-01

result with HHVM 3.4.2:
> 2015-01-08

**Comment by MartinMajor** [comment]
Any progress with this? It is quite dangerous issue, because it doesn't throw any exception or warning, it just returns incorrect data.

**Comment by Rattler3** [comment]
This is the same issue as #4681.

**Comment by MartinMajor** [comment]
This issue has been fixed in HHVM 3.8.

But there is still problematic behavior when invalid modifying string is used:

```php
$datetime = new DateTime('2015-01-08');
$datetime->modify('invalid string');
echo $datetime->format('Y-m-d');
```

It should throws exception or warning (like PHP).

**Comment by MartinMajor** [comment]
Great, thank you!
